sources/web/datalab/static/datalab.js

```diff
@@ -180,7 +194,8 @@ function initializePage(dialog, saveFn) {
       '<span class="fa fa-external-link-square">&nbsp;</span><a href="/static/about.txt" target="_blank">License and software information</a><br />' +
       '<span class="fa fa-external-link-square">&nbsp;</span><a href="https://cloud.google.com/terms/" target="_blank">Terms of Service</a><br />' +
       '<span class="fa fa-external-link-square">&nbsp;</span><a href="http://www.google.com/intl/en/policies/" target="_blank">Privacy Policy</a><br />' +
-      '<span class="fa fa-external-link-square">&nbsp;</span><a href="/static/reporting.html?enabled=' + reportingEnabled + '" target="_blank">Usage Statistics</a><br />';
+      '<span class="fa fa-external-link-square">&nbsp;</span><a href="/static/reporting.html?enabled=' + reportingEnabled + '" target="_blank">Usage Statistics</a><br />' +
+      '<span class="fa fa-recycle">&nbsp;</span><a href="javascript:restartDatalab()">Restart Server</a><br />';
```

Comment:
I'm hesitant to call it that, since it does not restart all of Datalab (e.g. the Docker container).

Instead, this just kills the NodeJS server, which then gets restarted by forever.

So, for instance, custom setup scripts will not re-run, nor will will it recreate things like SSH connections to a kernel gateway.